src/coreclr/jit/block.cpp

```diff
@@ -1630,59 +1630,6 @@ BasicBlock* BasicBlock::New(Compiler* compiler, BBjumpKinds jumpKind, unsigned j
     return block;
 }
 
-//------------------------------------------------------------------------
-// isBBCallAlwaysPair: Determine if this is the first block of a BBJ_CALLFINALLY/BBJ_ALWAYS pair
-//
-// Return Value:
-//    True iff "this" is the first block of a BBJ_CALLFINALLY/BBJ_ALWAYS pair
-//    -- a block corresponding to an exit from the try of a try/finally.
-//
-// Notes:
-//    In the flow graph, this becomes a block that calls the finally, and a second, immediately
-//    following empty block (in the bbNext chain) to which the finally will return, and which
-//    branches unconditionally to the next block to be executed outside the try/finally.
-//    Note that code is often generated differently than this description. For example, on ARM,
-//    the target of the BBJ_ALWAYS is loaded in LR (the return register), and a direct jump is
-//    made to the 'finally'. The effect is that the 'finally' returns directly to the target of
-//    the BBJ_ALWAYS. A "retless" BBJ_CALLFINALLY is one that has no corresponding BBJ_ALWAYS.
-//    This can happen if the finally is known to not return (e.g., it contains a 'throw'). In
-//    that case, the BBJ_CALLFINALLY flags has BBF_RETLESS_CALL set. Note that ARM never has
-//    "retless" BBJ_CALLFINALLY blocks due to a requirement to use the BBJ_ALWAYS for
-//    generating code.
-//
-bool BasicBlock::isBBCallAlwaysPair() const
```

Comment:
Moving methods from a `.cpp` file to a header is a takeback w.r.t. maintainability (the coding convention encourages placing methods in implementation files). Does it make a good enough difference to pay for that?

Reply:
Locally on Windows x64, I observed an additional 0.01% or 0.02% TP improvement; technically a 25-33% improvement over not force-inlining these methods, but only because the TP improvements were small on Windows x64 to begin with. As for maintainability, I agree I'd rather not move code previously in implementation files into header files, but I'm curious where we draw the line for ["very small inline member function implementations"](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md#9-function-structure) in the coding convention -- just one-line methods? Small methods with no function calls to avoid size increases from inlining?